cluster/schema/manager.go

```diff
@@ -377,3 +386,56 @@ func (s *SchemaManager) apply(op applyOp) error {
 	}
 	return nil
 }
+
+// migrateRangeIndexPropIfNeeded checks to see if any data which existed prior to 1.26
+// has been properly updated to reflect the new IndexRangeFilters property field which
+// was introduced in that version. Any data which existed prior to 1.26 will have a
+// value of `nil` for this field after upgrading to ≥ 1.26.
+//
+// The class migration flags are structured as a directory of files located in the raft
+// workdir, with each file mapping to a class name in the schema, on which the migration
+// has already run. If a classname file is not found, the migration will run once, and
+// set the flag, so it will never run again.
+func (s *SchemaManager) migrateRangeIndexPropIfNeeded(class *models.Class) error {
+	var (
+		rangeIdxMigrationPath = path.Join(s.workDir, "migration1.26.rangeIndexPropByClass")
+		classMigrationFlag    = path.Join(rangeIdxMigrationPath, class.Class)
+	)
+
+	// Has the migration ever been done for any class?
+	exists, err := diskio.FileExists(rangeIdxMigrationPath)
+	if err != nil {
+		return fmt.Errorf(
+			"check if rangeIdxMigrationPath (%s) exists: %w", rangeIdxMigrationPath, err)
+	}
+	if !exists {
+		if err := os.Mkdir(rangeIdxMigrationPath, os.ModePerm); err != nil {
+			return fmt.Errorf(
+				"mkdir rangeIdxMigrationPath (%s): %w", rangeIdxMigrationPath, err)
+		}
+	}
+
+	// Has the migration been done for this class specifically?
+	exists, err = diskio.FileExists(classMigrationFlag)
+	if err != nil {
+		return fmt.Errorf(
+			"check if class range index prop migration file (%s) exists: %w", classMigrationFlag, err)
+	}
+	if exists {
+		// Nothing to do, migration has already happened
+		return nil
+	}
+
+	// Run the migration and set the flag
+	for _, prop := range class.Properties {
+		if prop.IndexRangeFilters == nil {
```

Comment:
Can we move that condition at the top with an early exit clause ? It would allow us to skip all the FS operations. If we have lots of class these operations might be lengthy every time we reload the DB.

Reply:
decided to remove fs ops altogether, and just migrate on read